src/riverroute/mosart_physics_mod.F90

```diff
@@ -0,0 +1,604 @@
+MODULE MOSART_physics_mod
+
+   !-----------------------------------------------------------------------
+   ! Description: core code of MOSART.
+   ! Developed by Hongyi Li, 12/29/2011.
+   !-----------------------------------------------------------------------
+
+   use shr_kind_mod      , only : r8 => shr_kind_r8
+   use shr_const_mod     , only : SHR_CONST_REARTH, SHR_CONST_PI
+   use shr_sys_mod       , only : shr_sys_abort
+   use mosart_vars       , only : iulog, barrier_timers, mpicom_rof
+   use mosart_data       , only : Tctl, TUnit, TRunoff, TPara, ctl
+   use perf_mod          , only : t_startf, t_stopf
+   use nuopc_shr_methods , only : chkerr
+   use ESMF              , only : ESMF_FieldGet, ESMF_FieldSMM, ESMF_Finalize, &
+                                  ESMF_SUCCESS, ESMF_END_ABORT, ESMF_TERMORDER_SRCSEQ
+
+   implicit none
+   private
+
+   public :: Euler
+   public :: updatestate_hillslope
+   public :: updatestate_subnetwork
+   public :: updatestate_mainchannel
+   public :: hillsloperouting
+   public :: subnetworkrouting
+   public :: mainchannelrouting
+
+   real(r8), parameter :: TINYVALUE = 1.0e-14_r8 ! double precision variable has a significance of about 16 decimal digits
```

Comment:
use ISO_FORTRAN_ENV, only : REAL64
TINYVALUE = tiny(1._REAL64)

Reply:
This should probably be defined in shr_kind_mod.

Reply:
address now or open an issue?

Reply:
I have a lot on my plate right now, but if you want to open a PR to cesm_share I will happily review it.